tapgarden/batch.go

```diff
@@ -309,6 +315,35 @@ func (m *MintingBatch) HasSeedlings() bool {
 	return len(m.Seedlings) != 0
 }
 
+// ValidateSeedling checks if a seedling is valid for the batch.
+func (m *MintingBatch) ValidateSeedling(newSeedling Seedling) error {
```

Comment:
IIUC we want this to be a logical AND between this flag for each seedling, and the matching flag for the batch.

So the first seedling of the batch should update the batch flag, and every other seedling should match. This implies that every seedling would have `EnableUniAnnounce` set to `true`.

Reply:
Yes, that's what I'm going for. `EnableUniAnnounce` set to `true` on each seedling if it's true on the batch. They need to correspond basically. We can relax that sort of thing in the future. Let me know if you have thoughts on that sort of approach.